### PR TITLE
Slight performance improvement for RSA auth + MacOS build fix

### PR DESCRIPTION
**Pull request by KensingtonTech**
I've got another small update for you.  And this time I've got Docker set up on my build host so the .so is good.  Hopefully you should be able to just merge.

Removed memcpy op for RSA validation and use pointer instead.
build.sh fix for MacOS Docker.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -15,7 +15,9 @@ fi
 
 CONTAINER_ID=$(docker run --name "${DOCKER_IMAGE_NAME}-cont" -d -p 8000:8000 ${DOCKER_IMAGE_NAME})
 
-MACHINE_IP=`docker-machine ip`
+if ! MACHINE_IP=`docker-machine ip 2>/dev/null`; then
+  MACHINE_IP='0.0.0.0' # fix for MacOS
```
Comment (maxx-t) [inline build.sh]: Wouldn't it be better to use 127.0.0.1 or localhost instead of '0.0.0.0' ?

**Comment by KensingtonTech** [comment]
I admit that it’s strange and didn’t make sense to me, but that was the guidance from a Docker issue (have to find it), and it worked perfectly for me.

**Comment by KensingtonTech** [comment]
https://github.com/prakhar1989/docker-curriculum/issues/27

**Comment by kevinmichaelchen** [comment]
It's out of scope for this PR, but it'd probably be better to run the `curl` commands from inside a container and hit `http://host.docker.internal:8000`... 

e.g,.

```
docker run --rm byrnedo/alpine-curl --silent http://host.docker.internal:8000
```

https://stackoverflow.com/questions/24319662/from-inside-of-a-docker-container-how-do-i-connect-to-the-localhost-of-the-mach/24326540#24326540

**Pull request merged**